Documentation/docs/learn/faq.md

```diff
@@ -0,0 +1,611 @@
+Ruby). See the page about [ITK wrappings](https://itk.org/Wiki/ITK/Wrapping).
+
+### How do I use ITK from C# .NET?
+
+There are currently three options:
+
+1. Use [ManagedITK](https://code.google.com/archive/p/manageditk/)
+   - **NOTE: This option is not recommended as the project is no
+     longer supported**.
+1. Use the [SimpleITK](https://simpleitk.org/) C# language
+   bindings:
+   - SimpleITK is great, but it does not support all filters.
+   - A list of available filters can be found
+     [here](http://www.itk.org/SimpleITKDoxygen/html/Filter_Coverage.html)
+     (valid Sept 2012).
+1. Manually wrap the desired functionality
+   - This option requires the most work, but is the most flexible,
+     allowing you to use any ITK functionality you desire
+   - An example can be found [here](https://itk.org/Wiki/ITK/Using_ITK_from_.NET).
+
+NOTE: `WrapITK` does not currently support C# language bindings, but could
+be made to do so.
+
+## Using the Insight Toolkit
+
+### How do I combine ITK and VTK in my application?
+
+A full tutorial on how to combine ITK and VTK is available at [Getting Started with ITK and VTK](http://www.itk.org/CourseWare/Training/GettingStarted-II.pdf).
+
+You will need the auxiliary classes available at the [ITKApps](https://github.com/InsightSoftwareConsortium/ITKApps)
+repository.
+
+For example, for converting ITK images into VTK images and back, you
+will find useful the files from the [ITKVTKGlue](https://github.com/InsightSoftwareConsortium/ITKVtkGlue)
+module:
+
+```
+itkImageToVTKImageFilter.h
+itkImageToVTKImageFilter.hxx
+itkVTKImageToImageFilter.h
+itkVTKImageToImageFilter.hxx
+```
+
+### How do I configure Emacs for ITK coding style?
+
+You can add the following codes into your `.emacs` file:
+
+```
+;;;;;;;;;;;;;;
+(c-add-style "itk"
+            '("stroustrup"
+              (c-basic-offset . 2)
+              (c-offsets-alist
+                (c . c-lineup-dont-change)
+                (innamespace . 0)
+                (inline-open . 0)
+                (substatement-open . +)
+                (statement-block-intro . 0) ) ) )
+(add-hook 'c++-mode-hook (function (lambda ()
+ (c-set-style "itk")(turn-on-auto-fill))))
+(add-to-list 'auto-mode-alist '("\\.txx\\'" . c++-mode)) (add-to-list 'auto-mode-alist '("\\.h\\'" . c++-mode))
+;;;;;;;;;;;
+```
+
+### How do I configure Vim for ITK coding style?
+
+ITK provides a [`clang-format` configuration file](https://github.com/InsightSoftwareConsortium/ITK/blob/master/.clang-format)
+that ensures that the code being typed follows the ITK coding style.
+
+It suffices to tell Vim to use such configuration file.
+
+You can also use CMake indentation and syntax file which are available
+on www.vim.org. This was contributed by Andy Cedilnik while at Kitware.
+
+If you'd like to enable C++ style syntax highlight for `*.txx` files,
+put the following lines in your `.vimrc` file:
+
+```
+syntax on
+au BufNewFile,BufRead *.txx set filetype=cpp
+```
```

Comment:
Not sure if this is still relevant.

Reply:
It can be removed -- we changed .txx to .hxx.

Reply:
Removed in 894e5e6.